src/jobservice/job/impl/notification/http_helper.go

```diff
@@ -0,0 +1,41 @@
+package notification
+
+import (
+	"net/http"
+
+	commonhttp "github.com/goharbor/harbor/src/common/http"
+)
+
+const (
+	secure   = "secure"
+	insecure = "insecure"
+)
+
+var (
+	httpHelper *HTTPHelper
+)
+
+// HTTPHelper in charge of sending notification messages to remote endpoint
+type HTTPHelper struct {
```

Comment:
Do we really need this helper? Base on the scenario, it seems we just only need two singleton HTTP client functions.

```go
var insecHttpClient, secureHttpClient *http.Client

func getInsecuHttpClient() *http.Client{
  if insecHttpClient == nil {
      insecHttpClient = commonhttp.GetHTTPTransport(commonhttp.InsecureTransport)
  }

  return insecHttpClient
}

func getSecuHttpClient() *http.Client{
  if secureHttpClient == nil {
      secureHttpClient = commonhttp.GetHTTPTransport(commonhttp. SecureTransport)
  }

  return secureHttpClient
}
```

Reply:
we can define more httpclient of custom configurations. like connection pool settings, timeout settings and so on. and there are only two kinds now.

Reply:
This is something like over-engineering because we don't have other requirements till now (especially different timeout cases). If you like, we can keep it as it is now.